enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,509 @@
+last-updated: 2023-02-15
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPBU-4
+  - https://issues.redhat.com/browse/PORTENABLE-471
+  - https://issues.redhat.com/browse/CCO-286
+see-also:
+  - []
+replaces:
+  - []
+superseded-by:
+  - []
+---
+
+# Tokenized Authentication for Red Hat Operators on Cloud Providers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Amazon Web Services (AWS) Secure Token Service (STS) allows you to request temporary limited-privilege credentials. A concrete 
+example is write permission on an S3 bucket on AWS. STS is supported on all the major cloud providers where OpenShift is
+deployed. Currently, several Red Hat operators provide detailed instructions for how to use the operator on STS-enabled
+clusters. The process is designed to have manual operations performed by the customer. This enhancement seeks to bring common standards to the process of enabling an
+operator to access cloud provider resources, by enabling automation, UX and logic for the process. 
+
+## Motivation
+
+Several important Red Hat operators have normal operating modes where they assure their operands have required cloud 
+resources to perform properly. Currently, when a cluster is installed on a cloud platform an admin is required to 
+follow instructions (for example [1]) to provide STS-based credentials to the operator such that needed resources can be
+accessed by the operand. This enhancement seeks to normalize this process across operators, so users of several of them
+have the same experience and similar steps to perform. We aim to automate and reduce those steps as much as possible to 
+make adding operators (via OLM) simpler on an STS enabled cluster.
+
+### User Stories
+
+* As a cluster admin, I want to know which OLM Operators support tokenized authentication for my cloud, so that I can
+  provide token-based access for cloud resources for them.
+* As a cluster admin of a cluster using tokenized cloud auth, I want to know what's required to install and upgrade OLM 
+  Operators whenever those operators manage resources that authenticate against my cloud so they can function properly
+* As an Operator developer, I want to have a standard framework to define tokenized authentication requirements and
+  consume them, per supported cloud, so that my operator will work on token-based authentication clusters.
+* As an Operator Hub browser, I want to know which operators support tokenized cloud auth and on which clouds so I can
+  see only a filtered list of operators that will work on the given cluster.
```

Comment:
Are there two facets here?
1. tokenized cloud auth is relevant to this operator
2. tokenized cloud auth is supported by this operator

If an operator doesn't talk to a cloud API at all, tokenized cloud auth is irrelevant, so should it still show up in this filtered list?

Reply:
Also, is this filter view aware of the credential mode of the cluster? If my cluster is not using STS mode, will I still be presented with a "supports STS view"?

Reply:
on non-STS, no, the idea is not to see supports STS or anything like that in the UX.

 On the former comment, that was considered. If an operator has no cloud access reqs, there's no harm is also listing it as supporting STS.

Reply:
> On the former comment, that was considered. If an operator has no cloud access reqs, there's no harm is also listing it as supporting STS.

how are you going to make the determination?  I only see two options:

1) opt-in - operators must tag themselves as supporting STS (including operators that don't need creds at all)
2) opt-out - operators must tag themselves as not supporting STS

of those, only (1) seems reasonable (we wouldn't want to have operators imply they support STS that don't), so that means to achieve "If an operator has no cloud access reqs, there's no harm is also listing it as supporting STS." you need to get all those operators to tag themselves as supporting STS.  Which i'm skeptical about achieving.

Which is to say, i expect when you turn on the "STS-supported" filter, you're going to see a subset of the operators that actually work on STS mode.

Which i think means we need to name the tag/flag/filter carefully, something like "STS enabled", meaning the operator has explicit support for STS, and do not pick a tag/flag/filter name that implies that "if the operator isn't on this list, it won't work on an STS cluster" (because many operators will work on STS clusters, despite not being explicilty enabled for them, because they have no cred requirements)

Reply:
Ok added verbiage to the OperatorHub workflow noting the badge naming suggestion and reasoning.